Classes/common/CDTFetchChanges.m

```diff
@@ -79,12 +90,46 @@ - (void)main
                                                            options:&options
                                                             filter:nil
                                                             params:nil];
-        lastSequence = [self notifyChanges:changes startingSequence:lastSequence];
-    } while (changes.count > 0);
+        if (!thereIsALimit) {
+            // Limit not set. Get as many results as we can.
+            lastSequence = [self notifyChanges:changes startingSequence:lastSequence];
+
+            doLoop = (changes.count > 0);
+        } else if (mResultsLimit > 0) {
+            // There is a limit but we have not reached it yet.
+            lastSequence = [self notifyChanges:changes startingSequence:lastSequence];
+
+            if (changes.count == 0) {
```

Comment:
I wonder if it's possible to simplify the logic in these if/else branches. You have
```
lastSequence = [self notifyChanges:changes startingSequence:lastSequence];
```
in two branches (this one and the one above) and also in those same branches you have
```
doLoop = (changes.count > 0);
```
and 
```
if (changes.count == 0) {
  doLoop = NO;
}
```
which mean the same thing.